### PR TITLE
more accurate wall bonk

### PR DESCRIPTION
**Pull request by chilishit**
yeah

https://user-images.githubusercontent.com/73906828/201570651-680882ab-d773-411f-bb0d-fc0bf95bcf8b.mp4

### CODE REVIEW

**Comment by Fluvian** [comment]
What exactly does this change?

**Comment by chilishit** [comment]
the one used in tpp has way too many frames, it's supossed to be about one or two frames (as seen on this footage https://youtu.be/E4Wg1ueVbEw?t=8 and on the video i posted)

File: src/game/mario_actions_airborne.c [hunk 1]
```diff
@@ -384,8 +384,7 @@ u32 common_air_action_step(struct MarioState *m, u32 landAction, s32 animation,
             set_mario_animation(m, animation);
 
             if (m->forwardVel > 16.0f) {
-                set_mario_animation(m, MARIO_ANIM_START_WALLKICK);
-#ifdef VERSION_SH
+#if ENABLE_RUMBLE
```
Comment (Fluvian) [inline src/game/mario_actions_airborne.c:387]: Needs to be `VERSION_SH`, `ENABLE_RUMBLE` does not exist in this refresh.

File: src/game/mario_actions_airborne.c [hunk 2]
```diff
@@ -1305,18 +1304,15 @@ s32 act_air_hit_wall(struct MarioState *m) {
         return set_mario_action(m, ACT_SOFT_BONK, 0);
     }
 
-#ifdef AVOID_UB
-    return
-#endif
-        set_mario_animation(m, MARIO_ANIM_START_WALLKICK);
+    set_mario_animation(m, MARIO_ANIM_START_WALLKICK);
+    return 0;
```
Comment (Fluvian) [inline src/game/mario_actions_airborne.c:1308]: Remove this return statement as the original was bugged and never returned properly either.

**Review by Fluvian — CHANGES_REQUESTED** [review]

File: src/game/mario_actions_airborne.c (hunk 2 above)
Comment (iProgramMC) [reply src/game/mario_actions_airborne.c:1308]: > as the original was bugged and never returned properly either.
Which is why the bugged animation frame doesn't show up in the original SM64.

File: src/game/mario_actions_airborne.c (hunk 1 above)
Comment (chilishit) [reply src/game/mario_actions_airborne.c:387]: wouldn't it be better to remove this rumble statement all together?

File: src/game/mario_actions_airborne.c (hunk 2 above)
Comment (chilishit) [reply src/game/mario_actions_airborne.c:1308]: https://user-images.githubusercontent.com/73906828/204142213-b11d3a90-821a-48f5-9065-9193c0f2e2ba.mp4

just did what you told me to do, i was hoping it'd fix it but it actually messes it up

Comment (chilishit) [reply src/game/mario_actions_airborne.c:1308]: try it out by yourself if you can and you'll see what i mean with this

Comment (Fluvian) [reply src/game/mario_actions_airborne.c:1305]: I guess if that breaks it just revert that change, I'll accept it.

**Review by Fluvian — APPROVED** [review]

**Pull request merged**